src/legacy/ui/public/state_management/state_hashing/state_hash.ts

```diff
@@ -17,12 +17,13 @@
  * under the License.
  */
 
-import { Sha256 } from '../../../../../core/public/utils/';
+import { Sha256 } from '../../../../../core/public/utils';
+import { HashedItemStoreSingleton } from '../state_storage';
 
 // This prefix is used to identify hash strings that have been encoded in the URL.
 const HASH_PREFIX = 'h@';
 
-export function createStateHash(json, existingJsonProvider) {
```

Comment:
I didn't find a use case where this callback would actually make things simpler, so I decided to go fo HashedItemStoreSingleton directly here to simplify consumer code.

Reply:
Yeah good point; I think we'll have to change this anyway when we get rid of the singleton, but doesn't hurt to change in the interim for clarity.

Reply:
returned optional `existingJsonProvider` as state.js relies on it in tests